iree/test/e2e/models/mnist_inference.mlir

```diff
@@ -0,0 +1,122 @@
+// MNIST model with empty weights, for testing.
+//
+// Generated by the colab/mnist_training.ipynb notebook with manual edits to
+// * remove remaining TensorFlow attributes
+// * replace large values with zeros
+//
+// Note: batch size is set to 32
+
+// RUN: iree-run-mlir -iree-hal-target-backends=vmvx %s -function-input="32x28x28x1xf32" | IreeFileCheck %s
+// RUN: [[ $IREE_LLVMAOT_DISABLE == 1 ]] || (iree-run-mlir -iree-hal-target-backends=dylib-llvm-aot %s -function-input="32x28x28x1xf32" | IreeFileCheck %s)
+// RUN: [[ $IREE_VULKAN_DISABLE == 1 ]] || (iree-run-mlir -iree-hal-target-backends=vulkan-spirv %s -function-input="32x28x28x1xf32" | IreeFileCheck %s)
+
+#map0 = affine_map<(d0, d1) -> (d0, d1)>
+#map1 = affine_map<(d0, d1) -> ()>
+#map2 = affine_map<(d0, d1) -> (d0)>
+#map3 = affine_map<(d0, d1) -> (d0, 0)>
+module  {
+  util.global private @"__iree_flow___sm_node22__model.layer-2.kernel" = dense<0.000000e+00> : tensor<784x128xf32>
+  util.global private @"__iree_flow___sm_node23__model.layer-2.bias" = dense<0.000000e+00> : tensor<128xf32>
+  util.global private @"__iree_flow___sm_node32__model.layer-4.kernel" = dense<0.000000e+00> : tensor<128x10xf32>
+  util.global private @"__iree_flow___sm_node33__model.layer-4.bias" = dense<0.000000e+00> : tensor<10xf32>
+  // CHECK-LABEL: EXEC @predict
+  func @predict(%arg0: !hal.buffer_view) -> !hal.buffer_view attributes {iree.abi = "{\22a\22:[[\22ndarray\22,\22f32\22,4,32,28,28,1]],\22r\22:[[\22ndarray\22,\22f32\22,2,32,10]],\22v\22:1}"} {
```

Comment:
We probably want these one step sooner in the pipeline - ideally with iree.* ops (I'm not sure of the status there). That'll let us retarget to other binding layers as we have them (tflite/native today) from these same sources. The iree.abi attribute should be there but the hal.tensor.cast ops should not be.

We run the ABI pipeline at the head of translation: https://github.com/google/iree/blob/247a2a499e9112c2d1bcc81ee96017a95ecb4ed2/iree/compiler/Translation/IREEVM.cpp#L131
that's what decides the ABI, and we want that to be configurable.

Using the IREE dialect will make this file more stable (vs when I change hal.tensor.cast/util.global/etc).

Reply:
@stellaraccident what's the status of importing into this earlier translation stage? 

These are generated by this code right now
```python
iree.compiler.tf.compile_module(
    TrainableDNN(),
    import_only=True,
    output_mlir_debuginfo=False,
    output_file=inference_output_file,
    exported_names=["predict"])
```
I could switch to invoking `iree-import-tf` directly as needed too.

Reply:
I tried all of these: https://github.com/google/iree/blob/288ff986e361a250e10f898f842b6e5d6098e7ea/bindings/python/iree/compiler/tf.py#L84-L86

* `tf_input` is too early (full of `tf_executor.*` ops)
* `mid_level_input` (I also don't understand what that means: https://github.com/google/iree/pull/6439) is a weird mix - has `hal.tensor.cast`, `mhlo`, and `chlo`
* `iree_input` is the same as `output_file`

so something in https://github.com/google/iree/blob/main/integrations/tensorflow/iree_tf_compiler/TF/Passes.cpp is inserting `hal.tensor.cast` ops. I can run the tool directly with `-print-ir-after-all` to see where that happens... once I figure out how to convert that python call to a call to the native tool.